src/core/osd.c

```diff
@@ -335,7 +335,25 @@ void osd_channel_show(bool bShow)
 static void create_osd_object(lv_obj_t **obj, const char *img, int index)
 {
 	int x = 0;
+	int left_offset = 0;
+	int right_offset = 0;
+	int osd_embedded_mode;
+	char str[128];
+	
+	ini_gets("osd", "embedded_mode", "2", str, sizeof(str), SETTING_INI);
+	osd_embedded_mode = atoi(str);
```

Comment:
It would be better adding `embedded_mode` to `g_setting` structure and loading this one time in `main.c` where all the other initialisation settings are loaded. Also use `ini_getl` to load a number setting.

Reply:
moved but leaving as string for now as I may change values to "4x3" and "16x9" respectively.